examples/proposed/url-shortener.w

```diff
@@ -0,0 +1,141 @@
+bring cloud;
+
+let EMPTY_JSON = Json "empty";
+
+class Utils {
+  extern "./url-shortener.js" inflight makeId(): str;
+  extern "./url-shortener.js" inflight fetch(url: str, method: str, body: Json?): str;
+}
+let utils = new Utils();
+
+class UrlShortener {
+  urlLookup: cloud.Bucket;
+  idLookup: cloud.Bucket;
+
+  init() {
+    // map from urls to ids
+    this.urlLookup = new cloud.Bucket() as "UrlLookup";
+
+    // map from ids to urls
+    this.idLookup = new cloud.Bucket() as "IdLookup";
+  }
+
+  // Returns an id for the given long url. Creates a new id if one does not
+  // already exist.
+  inflight getId(url: str): str {
+    if this.urlLookup.exists(url) {
+      return this.urlLookup.get(url);
+    }
+  
+    let id = utils.makeId();
+    this.idLookup.put(id, url);
+    this.urlLookup.put(url, id);
```

Comment:
How do you handle a broken transaction?

Likely on the read path...

Reply:
Got it. OK I tweaked this method and documented it to fix the broken transaction issue (I think)